pkg/clusterlogging/clusterlogforwarder_test.go

```diff
@@ -0,0 +1,418 @@
+		assert.Nil(t, nil, testCase.testClusterLogForwarder.Object)
+		testCase.testClusterLogForwarder.WithOutput(&testCase.outputs)
+		_, err := testCase.testClusterLogForwarder.Update(true)
+
+		if testCase.expectedError != "" {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
+			assert.Equal(t, []clov1.OutputSpec{testCase.outputs},
+				testCase.testClusterLogForwarder.Definition.Spec.Outputs)
+		}
+
+		testCase.testClusterLogForwarder.WithPipeline(&testCase.pipelines)
+		_, err = testCase.testClusterLogForwarder.Update(true)
+
+		if testCase.expectedError != "" {
+			assert.Equal(t, testCase.expectedError, err.Error())
+		} else {
+			assert.Equal(t, []clov1.PipelineSpec{testCase.pipelines},
+				testCase.testClusterLogForwarder.Definition.Spec.Pipelines)
+		}
+	}
+}
+
+func TestWithOutput(t *testing.T) {
+	testCases := []struct {
+		testOutputs       *clov1.OutputSpec
+		expectedError     bool
+		expectedErrorText string
+	}{
+		{
+			testOutputs:       &newOutputs,
+			expectedError:     false,
+			expectedErrorText: "",
+		},
+		{
+			testOutputs:       nil,
+			expectedError:     true,
+			expectedErrorText: "'outputSpec' parameter is empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidClusterLogForwarderBuilder(buildClusterLogForwarderClientWithDummyObject())
+
+		result := testBuilder.WithOutput(testCase.testOutputs)
+
+		if testCase.expectedError {
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, result.errorMsg)
+			}
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, []clov1.OutputSpec{*testCase.testOutputs}, result.Definition.Spec.Outputs)
+		}
+	}
+}
+
+func TestWithPipeline(t *testing.T) {
+	testCases := []struct {
+		testPipeline      *clov1.PipelineSpec
+		expectedError     bool
+		expectedErrorText string
+	}{
+		{
+			testPipeline:      &newPipelines,
+			expectedError:     false,
+			expectedErrorText: "",
+		},
+		{
+			testPipeline:      nil,
+			expectedError:     true,
+			expectedErrorText: "'pipelineSpec' parameter is empty",
+		},
+	}
+
+	for _, testCase := range testCases {
+		testBuilder := buildValidClusterLogForwarderBuilder(buildClusterLogForwarderClientWithDummyObject())
+
+		result := testBuilder.WithPipeline(testCase.testPipeline)
+
+		if testCase.expectedError {
+			if testCase.expectedErrorText != "" {
+				assert.Equal(t, testCase.expectedErrorText, result.errorMsg)
+			}
+		} else {
+			assert.NotNil(t, result)
+			assert.Equal(t, []clov1.PipelineSpec{*testCase.testPipeline}, result.Definition.Spec.Pipelines)
+		}
+	}
+}
+
+func buildValidClusterLogForwarderBuilder(apiClient *clients.Settings) *ClusterLogForwarderBuilder {
+	clusterLogForwarderBuilder := NewClusterLogForwarderBuilder(
+		apiClient, defaultClusterLogForwarderName, defaultClusterLogForwarderNamespace)
+	clusterLogForwarderBuilder.Definition.ResourceVersion = "999"
+	clusterLogForwarderBuilder.Definition.TypeMeta = metav1.TypeMeta{
```

Comment:
You can safely remove it

Reply:
I can not
```shell
=== RUN   TestClusterLogForwarderCreate
    clusterlogforwarder_test.go:229: 
                Error Trace:    /home/elgerman/eco-gotest/eco-goinfra-ir/pkg/clusterlogging/clusterlogforwarder_test.go:229
                Error:          Not equal: 
                                expected: &v1.ClusterLogForwarder{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"instance", GenerateName:"", Namespace:"openshift-logging", SelfLink:"", UID:"", ResourceVersion:"999", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.ClusterLogForwarderSpec{Inputs:[]v1.InputSpec(nil), Outputs:[]v1.OutputSpec(nil), Filters:[]v1.FilterSpec(nil), Pipelines:[]v1.PipelineSpec(nil), ServiceAccountName:"", OutputDefaults:(*v1.OutputDefaults)(nil)}, Status:v1.ClusterLogForwarderStatus{Conditions:status.Conditions(nil), Inputs:v1.NamedConditions(nil), Outputs:v1.NamedConditions(nil), Filters:v1.NamedConditions(nil), Pipelines:v1.NamedConditions(nil)}}
                                actual  : &v1.ClusterLogForwarder{TypeMeta:v1.TypeMeta{Kind:"ClusterLogForwarder", APIVersion:"logging.openshift.io/v1"}, ObjectMeta:v1.ObjectMeta{Name:"instance", GenerateName:"", Namespace:"openshift-logging", SelfLink:"", UID:"", ResourceVersion:"999", Generation:0, CreationTimestamp:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), DeletionTimestamp:<nil>, DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Spec:v1.ClusterLogForwarderSpec{Inputs:[]v1.InputSpec(nil), Outputs:[]v1.OutputSpec(nil), Filters:[]v1.FilterSpec(nil), Pipelines:[]v1.PipelineSpec(nil), ServiceAccountName:"", OutputDefaults:(*v1.OutputDefaults)(nil)}, Status:v1.ClusterLogForwarderStatus{Conditions:status.Conditions(nil), Inputs:v1.NamedConditions(nil), Outputs:v1.NamedConditions(nil), Filters:v1.NamedConditions(nil), Pipelines:v1.NamedConditions(nil)}}
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -2,4 +2,4 @@
                                  TypeMeta: (v1.TypeMeta) {
                                -  Kind: (string) "",
                                -  APIVersion: (string) ""
                                +  Kind: (string) (len=19) "ClusterLogForwarder",
                                +  APIVersion: (string) (len=23) "logging.openshift.io/v1"
                                  },
                Test:           TestClusterLogForwarderCreate

```

Reply:
this was a WA do not fail after removing you ask me before

Reply:
@kononovn I see how you solve it for the clusterlogging, but why to compare just name and namespace if we can compare the object itself?